prog.go

```diff
@@ -707,83 +700,107 @@ type RunOptions struct {
 	// CPU to run Program on. Optional field.
 	// Note not all program types support this field.
 	CPU uint32
-	// Called whenever the syscall is interrupted, and should be set to testing.B.ResetTimer
-	// or similar. Typically used during benchmarking. Optional field.
-	//
-	// Deprecated: use [testing.B.ReportMetric] with unit "ns/op" instead.
-	Reset func()
+}
+
+type testingTB interface {
```

Comment:
nit: maybe add a comment that this interface intends to mimic the interface of the package testing without importing the package.